File: src/main/kotlin/me/night0721/nv/packets/handle/PacketInjector.java
package me.night0721.nv.packets.handle;

import io.netty.channel.Channel;
import org.bukkit.entity.Player;

public class PacketInjector {
    public void addPlayer(Player p) {
        try {
            Channel ch = me.night0721.nv.packets.protocol.Channel.getChannel(p);
            if (ch.pipeline().get("PacketInjector") == null) {
                PacketHandler h = new PacketHandler(p);
                ch.pipeline().addBefore("packet_handler", "PacketInjector", h);
            }
        } catch (Throwable t) {
            t.printStackTrace();
        }
    }

    public void removePlayer(Player p) {
        try {
            Channel ch = me.night0721.nv.packets.protocol.Channel.getChannel(p);
            if (ch.pipeline().get("PacketInjector") != null) {
                ch.pipeline().remove("PacketInjector");
            }
        } catch (Throwable t) {
            t.printStackTrace();
        }
    }

}

File: src/main/kotlin/me/night0721/nv/packets/handle/PacketInjector.kt
package me.night0721.nv.packets.handle

import me.night0721.nv.packets.protocol.Channel
import org.bukkit.entity.Player

class PacketInjector {
    fun addPlayer(p: Player) {
        try {
            val ch = Channel.getChannel(p)
            if (ch!!.pipeline()["PacketInjector"] == null) {
                val h = PacketHandler(p)
                ch.pipeline().addBefore("packet_handler", "PacketInjector", h)
            }
        } catch (t: Throwable) {
            t.printStackTrace()
        }
    }

    fun removePlayer(p: Player) {
        try {
            val ch = Channel.getChannel(p)
            if (ch!!.pipeline()["PacketInjector"] != null) {
                ch.pipeline().remove("PacketInjector")
            }
        } catch (t: Throwable) {
            t.printStackTrace()
        }
    }
}